src/app/api/users/route.ts

```diff
@@ -36,3 +38,49 @@ export async function GET() {
 
     return NextResponse.json(users, { status: 200 });
 }
+
+const schema = zfd.formData({
+    inviteToken: zfd.text(),
+    password: zfd.text()
+});
+
+/**
+ * Creates a new User record.
+ * Parameters are passed via form data.
+ * @param inviteToken Corresponds to token in existing UserInvite record
+ * @param password Password for new User account
+ * @returns 400 if bad form data or expired UserInvite
+ * @returns 404 if UserInvite does not exist
+ * @returns 409 if User record for corresponding UserInvite already exists
+ * @returns 200
+ */
+export async function POST(req: NextRequest) {
+    const parsed = schema.safeParse(await req.formData());
+    if (!parsed.success) {
+        return argumentError("Invalid user data");
+    }
+    
+    const { inviteToken, password } = parsed.data;
+    const userInvite = await db.userInvite.findUnique({
+        where: {
+            token: inviteToken
+        }
+    });
+    if (!userInvite) {
+        return notFoundError("Invite does not exist");
+    } else if (userInvite.expiration < new Date()) {
+        return argumentError("Invite has expired");
+    }
+    try {
+        await db.user.create({
+            data: {
+                email: userInvite.email,
+                passwordHash: await argon2.hash(password),
+                type: userInvite.userType
+            }
+        });
+    } catch {
+        return conflictError("User already exists");
+    }
```

Comment:
this catches all errors-- need to check if the error is a conflict error

Reply:
see https://www.prisma.io/docs/orm/prisma-client/debugging-and-troubleshooting/handling-exceptions-and-errors